packages/api/src/tx/msg.ts

```diff
@@ -0,0 +1,90 @@
+import { TxRaw } from 'cosmjs-types/cosmos/tx/v1beta1/tx';
+import { defaultRegistryTypes } from '@cosmjs/stargate';
+import { Registry } from '@cosmjs/proto-signing';
+
+import { ConnectionOptions } from '../api';
+import { createStargateSigningClient } from './stargate-signing';
+
+export interface MessageClient {
+  readonly sendTokens: (
+    amount: number,
+    sender: string,
+    recipient: string,
+  ) => Promise<Uint8Array>;
+  readonly broadcast: (signedTxBytes: Uint8Array) => Promise<string>;
+}
+
+export async function setupTxExtension(
+  connection: ConnectionOptions,
+): Promise<MessageClient> {
+  const registry = new Registry([
+    ...defaultRegistryTypes,
+    // ["/my.custom.MsgXxx", MsgXxx], // Replace with your own type URL and Msg class
+  ]);
+
+  const connectionOptions = {
+    ...connection,
+    registry,
+  };
+  const signingClient = await createStargateSigningClient(connectionOptions);
+
+  /**
+   * Sign a transaction for sending tokens to a recipient
+   */
+  const sendTokens = async (
```

Comment:
we should probably just expose a `sign` function instead, but I guess this was for the purpose of the example